### PR TITLE
Upgrade `kind` to v0.6.1

### LINKED ISSUE #3852: CI: Upgrade Kind version
Both in the linkerd2 repo and the build box

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #3852

Upgraded `/bin/kind` to pull v0.6.1.
Also have `workflow.yml` use `KUBECONFIG` explicitly for setting the
location of the config file, now that `kind get kubeconfig-path` has
been deprecated (check
https://github.com/kubernetes-sigs/kind/releases/tag/v0.6.0 for detailed
info).
Note that in the build server the kind binary for this version is
`kind-0.6.1`, leaving the `kind` binary still pointing to v0.5.1 while
this gets merged and all the PR branches get this.

### CODE REVIEW

**Comment by alpeb** [comment]
Apologies for the CI spam, but I gotta test this directly in the build server :pray:

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -308,17 +308,18 @@ jobs:
...
+          bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug --config=$CUSTOM_DOMAIN_CONFIG --image kindest/node:v1.15.6 ||
+            bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug --config=$CUSTOM_DOMAIN_CONFIG --image kindest/node:v1.15.6
         else
-          bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug ||
-            bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug
+          bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug --image kindest/node:v1.15.6 ||
+            bin/kind create cluster --name=$KIND_CLUSTER --wait=2m --loglevel debug --image kindest/node:v1.15.6
         fi
-        scp $(bin/kind get kubeconfig-path --name=$KIND_CLUSTER) linkerd-docker:/tmp
+        scp /tmp/kind-config-$KIND_CLUSTER linkerd-docker:/tmp
```
Comment (siggy) [inline .github/workflows/workflow.yml]: can you do:
```bash
scp $KUBECONFIG linkerd-docker:/tmp
```

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -390,9 +393,10 @@ jobs:
         # Restore kubeconfig from remote docker host.
         mkdir -p $HOME/.kube
         scp linkerd-docker:/tmp/kind-config-$KIND_CLUSTER $HOME/.kube
-        export KUBECONFIG=$(bin/kind get kubeconfig-path --name=$KIND_CLUSTER)
+        export KUBECONFIG=$HOME/.kube/kind-config-$KIND_CLUSTER
```
Comment (siggy) [inline .github/workflows/workflow.yml]: similar to previous comment:
```bash
export KUBECONFIG=$HOME/.kube/kind-config-$KIND_CLUSTER
scp linkerd-docker:/tmp/kind-config-$KIND_CLUSTER $KUBECONFIG
```

...or, going a bit further, it looks like there is a new subcommand to re-export a kubeconfig file without creating a cluster. it's possible you could remove the `scp` commands to copy the kubeconfig file back and forth, and instead do:
```bash
export KUBECONFIG=$HOME/.kube/kind-config-$KIND_CLUSTER
kind export kubeconfig --name=$KIND_CLUSTER
```

this is a bit beyond the scope of a kind version bump, so ok to explore in a subsequent PR.

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -356,10 +357,12 @@ jobs:
         export KIND_CLUSTER=github-$TAG-${{ matrix.integration_test }}
         ssh -T linkerd-docker > /dev/null << EOF
           # TODO: This is using the kind binary on the remote host.
-          kind load docker-image gcr.io/linkerd-io/proxy-init:v1.3.0 --name=$KIND_CLUSTER
-          kind load docker-image prom/prometheus:v2.11.1 --name=$KIND_CLUSTER
+          # TODO: 'kind' still points to v0.5.1. When there are no more old CI branches depending on that
+          # we can replace it with v0.6.1. In the meantime we explicitly use 'kind-0.6.1'
+          kind-0.6.1 load docker-image gcr.io/linkerd-io/proxy-init:v1.3.0 --name=$KIND_CLUSTER
+          kind-0.6.1 load docker-image prom/prometheus:v2.11.1 --name=$KIND_CLUSTER
```
Comment (siggy) [inline .github/workflows/workflow.yml]: 👍

**Pull request merged**